### ISSUE TITLE
`.rst` render

### ISSUE THREAD

**Issue by boarpig** [gogs/gogs#211]
I installed your official linux x64 binary on my desktop and set it up to use sqlite. Then as a first thing I tried migrating my github repo

https://github.com/boarpig/issue

but it seemed to load forever. It seems it cloned it fine but cannot open the page for it. I suspected it could be something with the size of the repository but cloning gogs repository works fine.

Then I wondered if it was the name of the repo "issue" but I tried making a copy with a different name but that didn't seem to work either. There's something wrong with that repository, that makes gogs load forever and not even give any sort of error.

**Comment by theY4Kman** [comment]
Any news on this feature?

**Comment by tboerger** [comment]
What about https://github.com/hhatto/gorst as a rst parser?

**Comment by Unknwon** [comment]
Thanks @tboerger missed this one, but seems WIP and need some development as well.

**Comment by tboerger** [comment]
Sure, but maybe even that is better than just display ascii :)

**Comment by Unknwon** [comment]
@tboerger I'll see how much the package has implemented!

**Comment by loongw** [comment]
如果暂时找不到完美的go rst渲染引擎，能否实现一个扩展机制：
可以通过配置，让特定的文件类型交由另一个http server来渲染。这样，我们就可以用Python来实现渲染后端，也能保证完美的语法兼容性。

**Comment by Unknwon** [comment]
@loongw sounds workable...

**Comment by nithinphilips** [comment]
Agreed with @loongw. [Hugo](https://github.com/spf13/hugo) does something similar for alternative markup formatting. For rst, They basically execute `rst2html` and read the STDOUT.

Here's the hugo method that implements the rst rendering:
https://github.com/spf13/hugo/blob/5d3705df16014c1404282234e30ef9d1c9be9c15/helpers/content.go#L488

`rst2html` is installed as part of the [docutils](https://pypi.python.org/pypi/docutils) package and can output HTML from rst input.

This is an easy way to support markup languages that do not yet have a Go implementation.

**Comment by BoboTiG** [comment]
:+1:

**Comment by sluedecke** [comment]
This is related to https://github.com/gogits/gogs/issues/2097 and an external renderer might be a solution for both.

**Comment by lenisko** [comment]
Would be great to have support for it

**Comment by Vorlif** [comment]
+1

**Comment by ngulden** [comment]
+1

**Comment by Red-Li** [comment]
+1

**Comment by tboerger** [comment]
Please use the builtin +1 function, it's annoying to receive such useless emails

**Comment by lannocc** [comment]
I recently did a quick-and-dirty hack on Gogs to add some basic .rst support for our own use, but maybe you will also find it useful:

https://github.com/AlphaGriffin/gogs

**Comment by kaeptmblaubaer1000** [comment]
You could also try calling the docutils API via https://github.com/sbinet/go-python

**Comment by JamesTheAwesomeDude** [comment]
@kaeptmblaubaer1000 and @loongw's suggestions both sound awesome.

I really like the idea of actually linking in to an "authoritative" library (provided there's not too much performance impact, to get that beautiful canoncal parsing), but honestly even just some FastCGI-style hack sounds like it wouldn't really lock us in too much, and in the meantime provide an immensely effective workaround until this is supported fully.

**Comment by jakedp** [comment]
Why not just display it as text? Instead of looking for some complicated solution there must be a simple one in place.
